src/find_library.cpp

```diff
@@ -76,6 +78,27 @@ std::string find_library_path(const std::string & library_name)
   std::string search_path = get_env_var(kPathVar);
   std::list<std::string> search_paths = split(search_path, kPathSeparator);
 
+#if !defined(_WIN32) && !defined(__APPLE__)
```

Comment:
Could you extract this logic into a helper method?

```c++
#if !defined(_WIN32) && !defined(__APPLE__)

// <-- explain what the function is doing, and why we're doing this here.
std::list<std::string> retrieve_rpath_paths() {
  std::list<std::string> search_paths;
  std::string search_path_runpath;

  const ElfW(Dyn) * dyn = _DYNAMIC;
  const ElfW(Dyn) * runpath = NULL;
  const char * strtab = NULL;

  for (; dyn->d_tag != DT_NULL; ++dyn) {
    if (dyn->d_tag == DT_RUNPATH) {
      runpath = dyn;
    } else if (dyn->d_tag == DT_STRTAB) {
      strtab = (const char *)dyn->d_un.d_val;
    }
  }

  if (strtab != NULL && runpath != NULL) {
    search_path_runpath = std::string(strtab + runpath->d_un.d_val);
    search_paths.splice(search_paths.cend(), split(search_path_runpath, kPathSeparator));
  }
  return search_paths;
}
#else
std::list<std::string> retrieve_rpath_paths() {
  return std::list<std::string>();
}
#endif
```

...and then you could call the function where the implementation is currently located.

Reply:
Ok will do. If you don't mind I'd call it `retrieve_runpath_paths()` since I'm reading out `RUNPATH` as `RPATH` is deprecated.